source/Calamari.Common/CalamariFlavourProgram.cs

```diff
@@ -53,12 +55,26 @@ protected virtual int Run(string[] args)
 
                 var builder = new ContainerBuilder();
                 ConfigureContainer(builder, options);
-                using (var container = builder.Build())
+
+                using var container = builder.Build();
+                container.Resolve<VariableLogger>().LogVariables();
+
+#if DEBUG
+                var waitForDebugger = container.Resolve<IVariables>().Get(KnownVariables.Calamari.WaitForDebugger);
+
+                if (string.Equals(waitForDebugger, "true", StringComparison.CurrentCultureIgnoreCase))
                 {
-                    container.Resolve<VariableLogger>().LogVariables();
+                    using var proc = Process.GetCurrentProcess();
+                    Log.Info($"Waiting for debugger to attach... (PID: {proc.Id})");
 
-                    return ResolveAndExecuteCommand(container, options);
+                    while (!Debugger.IsAttached)
+                    {
+                        Thread.Sleep(100);
```

Comment:
100ms feels like a very aggressive poll - let's change this to 1000ms - users won't notice the difference.